vmaas_sync/metrics_cyndi_test.go

```diff
@@ -0,0 +1,88 @@
+package vmaas_sync //nolint:golint,stylecheck
+
+import (
+	"app/base/core"
+	"app/base/database"
+	"app/base/utils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestCyndiSystemsCounts(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	counts, err := getCyndiCounts(refTime())
+	assert.Nil(t, err)
+	assert.Equal(t, 14, counts[lastUploadLast1D])
+	assert.Equal(t, 14, counts[lastUploadAll])
+}
+
+func TestUploadedCyndiSystemsCounts1D(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), 1)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 14, nSystems)
+}
+
+func TestUploadedCyndiSystemsCounts7D(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), 7)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 14, nSystems)
+}
+
+func TestUploadedCyndiSystemsCounts30D(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), 30)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 14, nSystems)
+}
+
+func TestUploadedCyndiSystemsCountsNoSystems(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	refTime := time.Date(2020, 9, 23, 10, 0, 0, 0, time.UTC)
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime, 30)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 0, nSystems)
+}
+
+func TestUploadedCyndiSystemsCountsAllSystems(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), -1)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 14, nSystems)
+}
+
+func TestCyndiTags(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	cyndiData, err := getCyndiData()
+	assert.Nil(t, err)
+	assert.Equal(t, 14, cyndiData.SystemsCount)
+	assert.Equal(t, 0, cyndiData.UniqueTags)
```

Comment:
It's not too good testing with no tags? Please, add some testing ones and check counting here. Actually there should be some tags already. So why do we expect 0 unique tags here?

Reply:
There are only data that have same tags, so no tags are unique, I will add some data, with unique ones.